data/mods/gen1stadium/scripts.ts

```diff
@@ -529,15 +528,40 @@ export const Scripts: ModdedBattleScriptsData = {
 			basePower = this.battle.clampIntRange(basePower, 1);
 
 			// We now check attacker's and defender's stats.
-			let level = pokemon.level;
-			let attacker = pokemon;
-			const defender = target;
-			if (move.useTargetOffensive) attacker = target;
-			let atkType: StatIDExceptHP = (move.category === 'Physical') ? 'atk' : 'spa';
-			const defType: StatIDExceptHP = (move.defensiveCategory === 'Physical') ? 'def' : 'spd';
-			if (move.useSourceDefensiveAsOffensive) atkType = defType;
-			let attack = attacker.getStat(atkType);
-			let defense = defender.getStat(defType);
+			let level = source.level;
+			const attacker =
+				(move.useBaseOffensiveStatAndBoosts && move.useBaseOffensiveStatAndBoosts.includes('target')) ? target : source;
+			const defender =
+				(move.useBaseDefensiveStatAndBoosts && move.useBaseDefensiveStatAndBoosts.includes('source')) ? source : target;
+
+			let atkType: AllStatIDs = move.category === 'Physical' ? 'atk' : 'spa';
+			if (move.useBaseOffensiveStatAndBoosts) {
+				atkType = move.useBaseOffensiveStatAndBoosts.substr(7) as AllStatIDs;
+			}
+			let defType: AllStatIDs = move.category === 'Physical' ? 'def' : 'spd';
+			if (move.useBaseDefensiveStatAndBoosts) {
+				defType = move.useBaseDefensiveStatAndBoosts.substr(7) as AllStatIDs;
+			}
+
+			let attack;
+			let defense;
+
+			if (atkType === 'hp') {
+				attack = attacker.maxhp;
+			} else if (atkType === 'currenthp') {
+				attack = attacker.hp;
+			} else {
+				attack = attacker.getStat(atkType);
+			}
+
+			if (defType === 'hp') {
+				defense = defender.maxhp;
+			} else if (defType === 'currenthp') {
+				defense = defender.hp;
+			} else {
+				defense = defender.getStat(defType);
+			}
```

Comment:
ditto here

Reply:
I assume you are referencing optional chaining? It will not work here due to what is needed to be done for the alternative cases. If I am just missing something please let me know. I feel like this is the better and easier to read solution to the problem.

Reply:
no, i mean the inline if statements

Reply:
Fixed, thank you.